azavea/echo-locator
# Label and optionally filter ECC neighborhoods

## Overview

Add back support for ECC neighborhoods and move filter control to profile:

 - Add back `ecc` field to data processing pipeline for both neighborhood information GeoJSON files
 - Display a label in the neighborhood list and detail view cards for Expanded Choice Communities.
 - Add account profile field for optionally filtering out non-ECC neighborhoods
 - Filter to only ECC neighborhoods on map and sidebar when profile option to do so is set


### Demo

The new profile option (defaults to unchecked):
![image](https://user-images.githubusercontent.com/960264/69658167-b1ced280-1049-11ea-99fa-3c737ab1e1d7.png)


With profile default to show both ECC and non-ECC neighborhoods, in 'car' mode:
![image](https://user-images.githubusercontent.com/960264/69658120-95cb3100-1049-11ea-9cc4-edd921dbbd23.png)

With profile set to only show ECC neighborhoods, still in 'car' mode:
![image](https://user-images.githubusercontent.com/960264/69658220-c6ab6600-1049-11ea-94f1-b27c60eaba24.png)

An ECC neighborhood detail view (labeled in sidebar):
![image](https://user-images.githubusercontent.com/960264/69658347-05412080-104a-11ea-96e8-a13039c1de53.png)

## Notes

Much of this came from restoring code removed in #177; in particular, the data processing changes didn't require modification. Since the checkbox is now on the user profile instead of the map page form, the extra `data` property is no longer needed.

The structure of the sidebar changed since #177, so I just stuck the sidebar ECC label under the ranking charts; it probably could use some styling and/or get moved to somewhere else.


## Testing Instructions

 * Go to a profile
 * "Only show Expanded Choice Communities?" field should show and default to unchecked
 * Map should only show ECC neighborhoods when profile box is checked
 * Map should show all neighborhoods when profile box is unchecked
 * Setting should persist on page refresh

## Bonus testing (not necessary to test above steps)

 * `update_data.sh` script in `neighborhood_data` directory should process `ecc` field as expected


Closes #270
